terraform.sh

```diff
@@ -25,6 +25,7 @@ vault_path=${vault_path:-""}
 vault_ttl=${vault_ttl:-"15m"}
 vault_aws_role=${vault_aws_role:-"admin"}
 vault_aws_iam=${vault_aws_iam:-"false"}
+vault_fetch_provider_creds=${vault_fetch_provider_creds:-"true"}
```

Comment:
what about simply `vault=true`? WDYT?

Reply:
Yep, as you like, the thing is that it's only impacting the fetch of credentials at the provider level. It doesn't prevent all Vault actions such as fetching a token with the app roles creds.

Reply:
True and the `VAULT_ADDR` already acts as a "flag" for activating vault... 🤔 hm

What is the use case of such changes then?

We could mage `vault_path` mandatory to be passed in parameter of the CLI and this variable would act as a flag to fetch provider creds?

Reply:
The use case is the following, you run the command from an EC2 instance that already has access onto the S3 bucket, although you want that instance to fetch creds on an alternative aws account that is authorized in Vault which will be done in TF.